crates/ursa-metrics/src/server.rs

```diff
@@ -0,0 +1,54 @@
+use crate::config::MetricsServiceConfig;
+use anyhow::Result;
+use axum::{http::StatusCode, routing::get, Extension, Router};
+use lazy_static::lazy_static;
+use metrics_exporter_prometheus::{PrometheusBuilder, PrometheusHandle};
+use prometheus::{Encoder, Registry, TextEncoder};
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tracing::info;
+
+lazy_static! {
+    pub static ref BITSWAP_REGISTRY: Arc<Registry> = Arc::new(Registry::new());
+}
+
+async fn get_ping_handler() -> (StatusCode, String) {
+    (StatusCode::OK, "pong".to_string())
+}
+
+fn setup_metrics_handler() -> PrometheusHandle {
+    PrometheusBuilder::new().install_recorder().unwrap()
+}
+
+async fn metrics_handler(handle: Extension<Arc<PrometheusHandle>>) -> (StatusCode, String) {
+    // ursa metrics
+    let mut metrics = handle.render();
+
+    // Collect metrics provided from bitswap and append them to the metrics string
+    let mut buffer = Vec::new();
+    TextEncoder::new()
+        .encode(&BITSWAP_REGISTRY.gather(), &mut buffer)
+        .unwrap();
+    if !buffer.is_empty() {
+        metrics.push_str(&String::from_utf8(buffer).unwrap());
+    }
+
+    (StatusCode::OK, metrics)
+}
+
```

Comment:
Could we consolidate the endpoints under one server?

Reply:
hmm, I think this could be possible, but it'd require a refactor of both rpc and metrics I think. I had similar thoughts, and I think it'd be a good future improvement. I can track it in a new issue

Reply:
Actually was able to set this up pretty easily. I can push this now

Reply:
https://github.com/fleek-network/ursa/pull/81/commits/ff6eb5ff97be3cde04aee8e3846939fd8af897dd